ansys/mapdl/core/mapdl.py

```diff
@@ -145,10 +144,34 @@ def __init__(self, loglevel='DEBUG', use_vtk=True, log_apdl=False,
 
         self._post = PostProcessing(self)
 
+    @property
```

Comment:
Would this be better as a function? I think it's already a property, so this might be a breaking change, but seeing `with` for a property looks quite odd to me, maybe it's also the lack of an `as thing` as well!

Reply:
I agree that it's a bit odd, I was trying for simplicity here:

```
with mapdl.non_interactive:
   mapdl.prep7()
   ...
```

vs a more standard approach:

```py
with mapdl.non_interactive() as ni:
    ni.prep7()
    ...
```

However, the former approach is non-standard and I'll agree that it has side effects (since it's effectively changing the state of the `mapdl` object).  Let's consider changing this in a future PR.